### PR TITLE
(PIE-1144) Create run_puppet_plan function

### PR DESCRIPTION
**Pull request by coreymbe**
# Summary
  * This commit adds a new function named `run_puppet_plan`.

# Detailed Description
  * Added new file `bin/ta_puppet_alert_actions/puppet_plan_action.py` with `run_puppet_plan` function to be used by the modalert helper.

### CODE REVIEW

File: TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    helper.log_error(rmessage['message'])
+  else:
+    rmessage['message'] = 'Something happened to plan {} on {} that we have no idea about'.format(plan_name,pe_console)
+    helper.log_error(rmessage['message'])
+
+  pie.hec.post_action(rmessage, pe_console, splunk_hec_url, puppet_action_hec_token)
+
+# this is our interactive load option
+# assumes you're running this library directly from the command line
+# cat example_alert.json | python $thisfile.py
```
Comment (actowery) [inline TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py:120]: this was probably copypastad from the example? - we should update the filenames to be accurate

**Review by actowery — APPROVED** [review]
lgtm, tiny comment nitpick

File: TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    helper.log_error("Timeout must be an integer, '{}' was provided instead".format(plan_timeout))
+
+ # Check if the user is configured with an RBAC token or Password:
+  if alert['global']['pe_token']:
+    auth_token = bolt_user_pass
+  else:
+    auth_token = pie.rbac.genauthtoken(bolt_user,bolt_user_pass,'TA-puppet-alert-actions',rbac_url, timeout=token_lifetime)
+
+  # note: parameters is expected as a text string, not json, so in sample alert json must be represented as:
+  # "plan_parameters": "{ \"target\": \"puppet.angrydome.org\", \"service\": \"puppet\"}"
```
Comment (actowery) [inline TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py:81]: what's an angrydome?

File: TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py (hunk 1 above)
Comment (coreymbe) [reply TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py]: Good call, I went ahead and updated the steps for manually testing this.

File: TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py (hunk 2 above)
Comment (coreymbe) [reply TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py]: Thats probably an old testing hostname. I went ahead and updated this with a generic example hostname.

**Comment by coreymbe** [comment]
Good catch @actowery 👍. This one should be good now.

**Review by actowery — APPROVED** [review]

**Pull request merged**